src/jit/codegenarm64.cpp

```diff
@@ -2848,14 +2845,14 @@ void CodeGen::genCodeForCmpXchg(GenTreeCmpXchg* treeNode)
     }
 
     // The following instruction includes a release half barrier
-    // TODO-ARM64-CQ Evaluate whether this is necessary
-    // https://github.com/dotnet/coreclr/issues/14346
     getEmitter()->emitIns_R_R_R(INS_stlxr, emitTypeSize(treeNode), exResultReg, dataReg, addrReg);
 
     getEmitter()->emitIns_J_R(INS_cbnz, EA_4BYTE, labelRetry, exResultReg);
 
     genDefineTempLabel(labelCompareFail);
 
+    instGen_MemoryBarrier(INS_BARRIER_ISH);
```

Comment:
Is it possible to use the CASAL instruction? It is spec'ed to be atomic so it seems like that by itself would be sufficient.

Reply:
I would also have to look at the spec, but `casal` is effectively the same as the loop above.  So the extra barrier might still be needed.

Reply:
I see